Getting-Started/Code/Debugging/Logging/index-v8.md

```diff
@@ -0,0 +1,250 @@
+* `/App_Data/Logs/UmbracoTraceLog.DELLBOOK.20181108.json`
+
+
+## Structured logging
+Serilog is a logging framework that allows us to do structured logging or write log messages using the message template format, allowing us to have a more detailed log message, rather than the traditional text message in a long txt file.
+
+
+```
+2018-11-12 08:34:50,419 [P27004/D2/T1] INFO   Umbraco.Core.Runtime.CoreRuntime - Booted. (4586ms) [Timing 9e76e5f]
+```
+
+Here is an example of the same log message represented as JSON, you can see here we have much more information that would allow us to search & filter logs based on these properties with an appropiate logging system.
+
+```json
+{
+  "@t": "2018-11-12T08:34:50.4190399Z",
+  "@mt": "{EndMessage} ({Duration}ms) [Timing {TimingId}]",
+  "EndMessage": "Booted.",
+  "Duration": 4586,
+  "TimingId": "9e76e5f",
+  "SourceContext": "Umbraco.Core.Runtime.CoreRuntime",
+  "ProcessId": 27004,
+  "ProcessName": "iisexpress",
+  "ThreadId": 1,
+  "AppDomainId": 2,
+  "AppDomainAppId": "LMW3SVC2ROOT",
+  "MachineName": "DELLBOOK",
+  "Log4NetLevel": "INFO ",
+  "HttpRequestNumber": 1,
+  "HttpRequestId": "557f45ba-0888-4216-8723-e226d795a2f7"
+}
+```
+
+To learn more about structured logging and message templates you can read more about it over on the https://messagetemplates.org website or alternatively watch this video from Serilog creator - https://www.youtube.com/watch?v=OhmNp8UPEEg
+
+## Writing to the log
+Umbraco writes log messages, but you are also able to use the Umbraco logger to write the log file as needed, so you can get further insights and details about your implementation.
+
+Here is a simple example of using the logger to write an Information message to the log which will contain one property of **Name** which will output the name variable that is passed into the method
+
+```csharp
+using Umbraco.Web.WebApi;
+using Umbraco.Core.Logging;
+
+namespace MyNamespace
+{
+    public class MyApiController : UmbracoApiController
+    {
+        public string GetSayHello(string name)
+        {
+            Logger.Info<MyApiController>("We are saying hello to {Name}", name);
+            return $"Hello {name}";
+        }
+    }
+}
+```
+
+The incorrect way to log the message would be use string interpolation or string concatanation such as
+
+```csharp
+//Do not use :(
+Logger.Info(typeof(MyApiController), $"We are saying hello to {name}");
+Logger.Info<MyApiController>($"We are saying hello to {name}");
+
+//Do not use :(
+Logger.Info(typeof(MyApiController), "We are saying hello to " + name);
+Logger.Info<MyApiController>("We are saying hello to " + name);
+```
+
+The above examples will write to the log file, however we will not get a seperate property logged with the message and we have no easy way to search for all log messages of this type.
+
+Where as the previous example we would be able to find all log messages that use the message template `We are saying hello to {Name}`
```

Comment:
would it help when skimming this information to have the same pattern as the //Do Not Use ;-(, to reinforce the correct way?

eg

// Do Use :)
Logger.Info<MyApiController>("We are saying hello to {Name}", name);

it's like I immediately see the wrong way, but then can't find the right way, (although it is there!)